### PR TITLE
Full state name 

### PR DESCRIPTION
**Pull request by edwingl10**
## Motivation and Context

Displays full state instead of just the abbreviation. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] User visible change (users will notice UI or functional changes)

## How Has This Been Tested?

- [ ] Existing or new tests cover my changes.
- [ ] Manually tested locally or on the Render PR Server.

## Previewing Changes

1. Click the `onrender.com` URL the **render `bot`** posted below.
2. add a scholarship
3. add a state requirement

### CODE REVIEW

File: src/components/ScholarshipCard.tsx [hunk 1]
```diff
@@ -174,7 +175,7 @@ export default function ScholarshipCard({
               </Typography>
               <DetailCardCell
                 label="State"
-                text={reqs?.states?.join(', ') || 'All'}
+                text={reqs?.states?.map(State.toString).join(', ') || 'All'}
```
Comment (josueetcom) [inline src/components/ScholarshipCard.tsx:178]: Can you sort these while you're at it?

File: src/types/States.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+const toStringMappings: Record<string, string> = States.reduce(
+  (arr, s) => ({
+    ...arr,
+    [s.abbr]: `${s.name} (${s.abbr})`,
+  }),
+  {}
+);
+
+export namespace State {
+  export interface StateType {
```
Comment (josueetcom) [inline src/types/States.ts:72]: Is this not possible?
```ts
interface State {}
namespace State {}
export default State;
```

Comment (edwingl10) [reply src/types/States.ts:72]: yes! I dont know what I was missing but it was acting weird previously.

Comment (josueetcom) [reply src/types/States.ts]: Maybe you were exporting State twice.

File: src/types/States.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+Wisconsin	WI
+Wyoming	WY`;
+
+interface State {
+  /** Full name of the state or territory, e.g. `'Alabama'`. */
+  name: string;
+  /** State abbreviation, e.g. `'AL'`. */
+  abbr: string;
+}
+export const States: State[] = RAW_STATES.split('\n')
```
Comment (josueetcom) [inline src/types/States.ts:65]: Add a separating line and upper case the exported constant:
```suggestion

export const STATES: State[] = RAW_STATES.split('\n')
```

File: src/types/States.ts (hunk 2 above)
Comment (edwingl10) [reply src/types/States.ts]: yeah that was the problem 😂

**Review by josueetcom — APPROVED** [review]

File: src/types/States.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  {}
+);
+
+namespace State {
+  export function toString(state: string): string {
+    return toStringMappings[state];
+  }
+}
+
+export default State;
```
Comment (josueetcom) [inline src/types/States.ts:84]: Oh I forgot to mention that since the default exported name is State then they should go in the file called State.js

**Pull request merged**